### PR TITLE
1st user story

### PR DESCRIPTION
**Pull request by alevel-petrenko**

### CODE REVIEW

File: User_Story#1/SortingApp/SortinfApp.UnitTests/SorterFactory/InsertionSortCreatorTest.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    /// <owner>Anton Petrenko</owner>
+    [TestClass]
+    public class InsertionSortCreatorTest
+    {
+        /// <summary>
+        /// Tests Create method if call method it will get new instance InsertionSorter.
+        /// </summary>
+        /// <owner>Anton Petrenko</owner>
+        [TestMethod]
+        public void InsertionSortCreator_Create_CallMethod_GetNewInstanceInsertionSorter()
```
Comment (AlEkSunder) [inline User_Story#1/SortingApp/SortinfApp.UnitTests/SorterFactory/InsertionSortCreatorTest.cs]: "CallMethod" is excessive.

File: User_Story#1/SortingApp/SortinfApp.UnitTests/SorterFactory/QuickSortCreatorTest.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    /// <owner>Anton Petrenko</owner>
+    [TestClass]
+    public class QuickSortCreatorTest
+    {
+        /// <summary>
+        /// Tests Create method if call method it will get new instance QuickSorter.
+        /// </summary>
+        /// <owner>Anton Petrenko</owner>
+        [TestMethod]
+        public void QuickSortCreator_Create_CallMethod_GetNewInstanceSelectionSort()
```
Comment (AlEkSunder) [inline User_Story#1/SortingApp/SortinfApp.UnitTests/SorterFactory/QuickSortCreatorTest.cs]: CallMethod is excessive, please fix everywhere.

File: User_Story#1/SortingApp/SortinfApp.UnitTests/SortingAlgorithms/InsertionSorterTest.cs [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        /// Tests Sort method if pass sorted collection it will leave sorted collection.
+        /// </summary>
+        /// <owner>Anton Petrenko</owner>
+        [TestMethod]
+        public void InsertionSorter_Sort_PassSortedCollection_GetSortedCollection()
+        {
+            //
+            // Arrange.
+            //
+            var actualArray = new double[] { 0.15, 0.2, 1.56, 5.0, 9.99 };
```
Comment (AlEkSunder) [inline User_Story#1/SortingApp/SortinfApp.UnitTests/SortingAlgorithms/InsertionSorterTest.cs]: Why do you have the same values in array in the same order? What are you sorting? :)

File: User_Story#1/SortingApp/SortinfApp.UnitTests/SortingAlgorithms/InsertionSorterTest.cs [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+            //
+            Assert.ThrowsException<ArgumentNullException>(() => insertionSorter.Sort(array));
+        }
+
+        /// <summary>
+        /// Tests Sort method if pass sorted collection it will leave sorted collection.
+        /// </summary>
+        /// <owner>Anton Petrenko</owner>
+        [TestMethod]
+        public void InsertionSorter_Sort_PassSortedCollection_GetSortedCollection()
```
Comment (AlEkSunder) [inline User_Story#1/SortingApp/SortinfApp.UnitTests/SortingAlgorithms/InsertionSorterTest.cs:39]: I believe this logic is not valid for our test case... but let it be.

File: User_Story#1/SortingApp/SortinfApp.UnitTests/Utilities/Parser/ArrayParserTest.cs [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+    [TestClass]
+    public class ArrayParserTest
+    {
+        /// <summary>
+        /// Tests ConvertData if pass null content it will throw ArgumentNullException.
+        /// </summary>
+        /// <owner>Anton Petrenko</owner>
+        [TestMethod]
+        public void ArrayParser_ConvertData_PassEmptyContent_ThrowArgumentNullException()
+        {
```
Comment (AlEkSunder) [inline User_Story#1/SortingApp/SortinfApp.UnitTests/Utilities/Parser/ArrayParserTest.cs:21]: Missed method that checks if content is null.

File: User_Story#1/SortingApp/BusinessLayer/Utilities/Parser/ArrayParser.cs [hunk 6]
```diff
@@ -35,6 +35,9 @@ public ArrayParser()
         /// <param name="content">Content of the file.</param>
         public T[] ConvertData(string content)
         {
+            if (string.IsNullOrWhiteSpace(content))
+                throw new ArgumentNullException(nameof(content));
+
             try
```
Comment (AlEkSunder) [inline User_Story#1/SortingApp/BusinessLayer/Utilities/Parser/ArrayParser.cs:41]: Why do we have a try-catch here?

File: User_Story#1/SortingApp/SortinfApp.UnitTests/Utilities/Validator/LocalFileValidatorTest.cs [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+    /// <owner>Anton Petrenko</owner>
+    [TestClass]
+    public class LocalFileValidatorTest
+    {
+        /// <summary>
+        /// Tests IsDataExist if pass empty path it will throw ArgumentNullException.
+        /// </summary>
+        /// <owner>Anton Petrenko</owner>
+        [TestMethod]
+        public void LocalFileValidator_IsDataExist_PassEmtptyPath_ThrowArgumentNullException()
```
Comment (AlEkSunder) [inline User_Story#1/SortingApp/SortinfApp.UnitTests/Utilities/Validator/LocalFileValidatorTest.cs]: Incorrect description.
In this case path is null, not empty.

Comment (AlEkSunder) [reply User_Story#1/SortingApp/SortinfApp.UnitTests/Utilities/Validator/LocalFileValidatorTest.cs]: A separate test method should be added for check if path is empty.

File: User_Story#1/SortingApp/SortinfApp.UnitTests/Utilities/Validator/LocalFileValidatorTest.cs [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+            // Arrange.
+            //
+            var localFileValidator = new LocalFileValidator();
+            string path = @"D:\collectionToRead.txt";
+            bool actualResult;
+
+            //
+            // Act.
+            //
+            using (File.Create(path)) { }
```
Comment (AlEkSunder) [inline User_Story#1/SortingApp/SortinfApp.UnitTests/Utilities/Validator/LocalFileValidatorTest.cs:70]: We need to check if path is valid before we creating the file.
Directory.Exists(path)

File: User_Story#1/SortingApp/SortinfApp.UnitTests/Utilities/Validator/LocalFileValidatorTest.cs [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+        /// </summary>
+        /// <owner>Anton Petrenko</owner>
+        [TestMethod]
+        public void LocalFileValidator_IsDataExist_PassExistFile_GetTrue()
+        {
+            //
+            // Arrange.
+            //
+            var localFileValidator = new LocalFileValidator();
+            string path = @"D:\collectionToRead.txt";
```
Comment (AlEkSunder) [inline User_Story#1/SortingApp/SortinfApp.UnitTests/Utilities/Validator/LocalFileValidatorTest.cs]: I'd suggest to use user temporary directory.
Path.GetTempPath()

File: User_Story#1/SortingApp/SortinfApp.UnitTests/Utilities/Validator/СloudFileValidatorTest.cs [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+    /// <owner>Anton Petrenko</owner>
+    [TestClass]
+    public class СloudFileValidatorTest
+    {
+        /// <summary>
+        /// Tests IsDataExist if pass empty url it will throw ArgumentNullException.
+        /// </summary>
+        /// <owner>Anton Petrenko</owner>
+        [TestMethod]
+        public void СloudFileValidator_IsDataExist_PassEmtptyUrl_ThrowArgumentNullException()
```
Comment (AlEkSunder) [inline User_Story#1/SortingApp/SortinfApp.UnitTests/Utilities/Validator/СloudFileValidatorTest.cs]: URI also could be empty and null - these are separate tests.

File: User_Story#1/SortingApp/SortinfApp.UnitTests/Utilities/DataReaderTest.cs [hunk 11]
```diff
@@ -0,0 +1,118 @@
...
+    /// <owner>Anton Petrenko</owner>
+    [TestClass]
+    public class DataReaderTest
+    {
+        /// <summary>
+        /// Tests DataReader if pass empty validator in constructor it will throw ArgumentNullException.
+        /// </summary>
+        /// <owner>Anton Petrenko</owner>
+        [TestMethod]
+        public void DataReader_DataReader_PassEmptyValidatorInConstructor_ThrowArgumentNullException()
```
Comment (AlEkSunder) [inline User_Story#1/SortingApp/SortinfApp.UnitTests/Utilities/DataReaderTest.cs:21]: Please do not hesitate to add Arrange block for all tests.

File: User_Story#1/SortingApp/SortinfApp.UnitTests/Utilities/DataReaderTest.cs [hunk 12]
```diff
@@ -0,0 +1,118 @@
...
+        /// <owner>Anton Petrenko</owner>
+        [TestMethod]
+        public void DataReader_ReadContent_PassEmptyPathToTheFile_ThrowArgumentNullException()
+        {
+            //
+            // Arrange.
+            //
+            var validator = new StubIValidator()
+            {
+                IsDataExistString = (string str) => throw new ArgumentNullException(nameof(str))
```
Comment (AlEkSunder) [inline User_Story#1/SortingApp/SortinfApp.UnitTests/Utilities/DataReaderTest.cs]: In such case you have two asserts in a single test method.
Asserts should be separated.

File: User_Story#1/SortingApp/SortinfApp.UnitTests/Utilities/DataReaderTest.cs [hunk 13]
```diff
@@ -0,0 +1,118 @@
...
+        /// </summary>
+        /// <owner>Anton Petrenko</owner>
+        [TestMethod]
+        public void DataReader_ReadContent_PassIncorrectPathToTheFile_GetEmptyString()
+        {
+            //
+            // Arrange.
+            //
+            string actualResult;
+            var expectedResult = String.Empty;
```
Comment (AlEkSunder) [inline User_Story#1/SortingApp/SortinfApp.UnitTests/Utilities/DataReaderTest.cs]: please use alias string instead of type String.

File: User_Story#1/SortingApp/SortinfApp.UnitTests/Utilities/DataReaderTest.cs (hunk 12 above)
Comment (AlEkSunder) [reply User_Story#1/SortingApp/SortinfApp.UnitTests/Utilities/DataReaderTest.cs]: Please write me if you have questions.

File: User_Story#1/SortingApp/BusinessLayer/Utilities/DataReader.cs [hunk 14]
```diff
@@ -1,6 +1,6 @@
 ﻿using System;
-using BusinessLayer.Utilities.Validator.Interfaces;
 using System.IO;
+using BusinessLayer.Utilities.Validator.Interfaces;
 
 namespace BusinessLayer.Reader
 {
```
Comment (AlEkSunder) [inline User_Story#1/SortingApp/BusinessLayer/Utilities/DataReader.cs:6]: Since the Path property  is used in one methos, please move it as parameter in the method:
 ReadContent(string path)

File: User_Story#1/SortingApp/SortinfApp.UnitTests/Utilities/DataWriterTest.cs [hunk 15]
```diff
@@ -0,0 +1,98 @@
...
+    /// <owner>Anton Petrenko</owner>
+    [TestClass]
+    public class DataWriterTest
+    {
+        /// <summary>
+        /// Tests WriteContent if pass empty collection it will throw ArgumentNullException.
+        /// </summary>
+        /// <owner>Anton Petrenko</owner>
+        [TestMethod]
+        public void DataWriter_WriteContent_PassEmptyCollection_ThrowArgumentNullException()
```
Comment (AlEkSunder) [inline User_Story#1/SortingApp/SortinfApp.UnitTests/Utilities/DataWriterTest.cs:22]: You are checking if collection is null, not empty.
Please don't be confused with these terms.

Comment (AlEkSunder) [reply User_Story#1/SortingApp/SortinfApp.UnitTests/Utilities/DataWriterTest.cs:22]: One more test method need to be added to check if collection is empty.

File: User_Story#1/SortingApp/SortinfApp.UnitTests/CollectionSortHandlerTest.cs [hunk 16]
```diff
@@ -0,0 +1,388 @@
...
+        /// Collections the sort handler collection sort handler pass null value in constructor throw argument null exception.
+        /// </summary>
+        /// <owner>Anton Petrenko</owner>
+        [TestMethod]
+        public void CollectionSortHandler_CollectionSortHandler_PassNullParserInConstructor_ThrowArgumentNullException()
+        {
+            //
+            // Arrange.
+            //
+            var reader = new StubDataReader(new StubIValidator());
```
Comment (AlEkSunder) [inline User_Story#1/SortingApp/SortinfApp.UnitTests/CollectionSortHandlerTest.cs]: We need to separate Stubs and Shims.
Stub is used for abstractions only (abstract class and Interface)
Shim is used for instances.

Comment (AlEkSunder) [reply User_Story#1/SortingApp/SortinfApp.UnitTests/CollectionSortHandlerTest.cs]: Also don't what parameter you pass as null.
This should be specified in Arrange block.

File: User_Story#1/SortingApp/SortinfApp.UnitTests/CollectionSortHandlerTest.cs [hunk 17]
```diff
@@ -0,0 +1,388 @@
...
+            //
+            // Arrange.
+            //
+            var parser = new StubIDataParser<double>();
+            var writer = new StubDataWriter<double>();
+
+            //
+            // Assert.
+            //
+            Assert.ThrowsException<ArgumentNullException>(() => new CollectionSortHandler<double>(null, writer, parser));
```
Comment (AlEkSunder) [inline User_Story#1/SortingApp/SortinfApp.UnitTests/CollectionSortHandlerTest.cs]: The same about null.
Don't hesitate to add variable with name and initialize it with null.

File: User_Story#1/SortingApp/SortinfApp.UnitTests/CollectionSortHandlerTest.cs [hunk 18]
```diff
@@ -0,0 +1,388 @@
...
+        [TestMethod]
+        public void CollectionSortHandler_Execute_LeftEmptyUnsortedCollection_ThrowException()
+        {
+            using (ShimsContext.Create())
+            {
+                //
+                // Arrange.
+                //
+                var parser = new StubIDataParser<double>();
+                var reader = new ShimDataReader();
```
Comment (AlEkSunder) [inline User_Story#1/SortingApp/SortinfApp.UnitTests/CollectionSortHandlerTest.cs:104]: When we use Shim, we need to create ShimContext:
ShimContext.Create().
Also don't forget to cover with using.

File: User_Story#1/SortingApp/SortinfApp.UnitTests/CollectionSortHandlerTest.cs [hunk 19]
```diff
@@ -0,0 +1,388 @@
...
+            {
+                //
+                // Arrange.
+                //
+                var expectedCollection = new double[] { 1.5, 4, 11, 56, 166 };
+                var sorterType = SortType.InsertionSort;
+                var path = @"D:\collectionToRead.txt";
+                var parser = new StubIDataParser<double>()
+                {
+                    ConvertDataString = (str) => new double[] { 166, 11, 56, 4, 1.5 }
```
Comment (AlEkSunder) [inline User_Story#1/SortingApp/SortinfApp.UnitTests/CollectionSortHandlerTest.cs]: I'd store the array in a separate variable with name actualCollection.

File: User_Story#1/SortingApp/SortinfApp.UnitTests/CollectionSortHandlerTest.cs [hunk 20]
```diff
@@ -0,0 +1,388 @@
...
+                };
+                var reader = new ShimDataReader();
+                var writer = new StubDataWriter<double>();
```
Comment (AlEkSunder) [inline User_Story#1/SortingApp/SortinfApp.UnitTests/CollectionSortHandlerTest.cs]: Why do you have reader as Shim and in the same time use Stub for writer?

File: User_Story#1/SortingApp/SortinfApp.UnitTests/CollectionSortHandlerTest.cs [hunk 21]
```diff
@@ -0,0 +1,388 @@
...
+                CollectionSortHandler<double> collectionSortHandler;
+                bool result;
```
Comment (AlEkSunder) [inline User_Story#1/SortingApp/SortinfApp.UnitTests/CollectionSortHandlerTest.cs]: Please renamae result to something more understandable, like isCollectionSorted.

File: User_Story#1/SortingApp/SortinfApp.UnitTests/CollectionSortHandlerTest.cs [hunk 22]
```diff
@@ -0,0 +1,388 @@
...
+            using (ShimsContext.Create())
+            {
+                //
+                // Arrange.
+                //
+                CollectionSortHandler<int> collectionSortHandler;
+                var path = @"D:\collectionToRead.txt";
+                var parser = new StubIDataParser<int>()
+                {
+                    ConvertDataString = (str) => throw new ArgumentNullException()
```
Comment (AlEkSunder) [inline User_Story#1/SortingApp/SortinfApp.UnitTests/CollectionSortHandlerTest.cs]: Here are two asserts for the single test method. 
Should be also separated in all such cases.

File: User_Story#1/SortingApp/SortinfApp.UnitTests/CollectionSortHandlerTest.cs [hunk 23]
```diff
@@ -0,0 +1,388 @@
...
+
+                //
+                // Act.
+                //
+                collectionSortHandler = new CollectionSortHandler<int>(reader, writer, parser);
+
+                //
+                // Assert.
+                //
+                Assert.ThrowsException<ArgumentNullException>(() => collectionSortHandler.Read("   "));
```
Comment (AlEkSunder) [inline User_Story#1/SortingApp/SortinfApp.UnitTests/CollectionSortHandlerTest.cs]: Please exclude literal to a variable.

File: User_Story#1/SortingApp/ViewModel/CollectionSortingViewModel.cs [hunk 24]
```diff
@@ -153,6 +153,7 @@ private void ReadArray(object obj)
 
                 foreach (var number in this.handler.UnSortedCollection)
                     this.UnSortedCollectionOfNumbers.Add(number);
+
```
Comment (AlEkSunder) [inline User_Story#1/SortingApp/ViewModel/CollectionSortingViewModel.cs:159]: Great :)

File: User_Story#1/SortingApp/ViewModel/CollectionSortingViewModel.cs [hunk 25]
```diff
@@ -202,7 +203,12 @@ private void SaveArray(object obj)
         /// <owner>Anton Petrenko</owner>
         private void ShowDialogWindow()
         {
-            using (var openFileDialog = new OpenFileDialog())
+            using (var openFileDialog = new OpenFileDialog()
+            {
+                FileName = "Select a text file",
+                Filter = "Text files (*.txt)|*.txt",
+                Title = "Open text file"
+            })
```
Comment (AlEkSunder) [inline User_Story#1/SortingApp/ViewModel/CollectionSortingViewModel.cs:214]: Please look at all properties with setter and make sure you have a check if value is equal to the cached value in field.
It is very important on VM side to avoid notifying UI without real changes.

File: User_Story#1/SortingApp/ViewModel/CollectionSortingViewModel.cs [hunk 26]
```diff
@@ -153,6 +153,7 @@ private void ReadArray(object obj)
 
                 foreach (var number in this.handler.UnSortedCollection)
                     this.UnSortedCollectionOfNumbers.Add(number);
+
                 this.Message = "Array was successfully read.";
             }
             catch (Exception e)
```
Comment (AlEkSunder) [inline User_Story#1/SortingApp/ViewModel/CollectionSortingViewModel.cs:162]: If you had a dependency injection for VM, you would create tests for VM too.
So just to be clear: VM should be covered for Unit tests too.
But I believe it will be module testing, anyway it is a part of unit tests)

**Review by AlEkSunder — APPROVED** [review]
In general I see you succeeded with Unit testing.
But I'd suggest you:
1) to look at Shim/Stub difference;
2) to store null values in variable before you pass it as parameter;
3) to add separate tests for empty and null strings/collections;
4) to add add 'Arrange' block even if it contains a single line of code.

File: User_Story#1/SortingApp/SortinfApp.UnitTests/SortingAlgorithms/InsertionSorterTest.cs (hunk 4 above)
Comment (alevel-petrenko) [reply User_Story#1/SortingApp/SortinfApp.UnitTests/SortingAlgorithms/InsertionSorterTest.cs:39]: as a variant, I could read sorted array from file and I want to be sure that logic of my Sort method doesn't compromise the output.

File: User_Story#1/SortingApp/SortinfApp.UnitTests/Utilities/Validator/LocalFileValidatorTest.cs (hunk 8 above)
Comment (alevel-petrenko) [reply User_Story#1/SortingApp/SortinfApp.UnitTests/Utilities/Validator/LocalFileValidatorTest.cs:70]: Do you mean that such path could be not valid for some user and I should prevent creating file in not valid place? But in this case result of test case would be not pass too as when path not valid.

File: User_Story#1/SortingApp/SortinfApp.UnitTests/Utilities/DataReaderTest.cs (hunk 12 above)
Comment (alevel-petrenko) [reply User_Story#1/SortingApp/SortinfApp.UnitTests/Utilities/DataReaderTest.cs]: I've slightly changed test case in Arrange part, but I don't understand how to avoid using both of these initialization.

File: User_Story#1/SortingApp/SortinfApp.UnitTests/CollectionSortHandlerTest.cs (hunk 22 above)
Comment (alevel-petrenko) [reply User_Story#1/SortingApp/SortinfApp.UnitTests/CollectionSortHandlerTest.cs]: I've slightly changed test case in Arrange part and changed name please check again.

File: User_Story#1/SortingApp/BusinessLayer/Utilities/Parser/ArrayParser.cs (hunk 6 above)
Comment (alevel-petrenko) [reply User_Story#1/SortingApp/BusinessLayer/Utilities/Parser/ArrayParser.cs:41]: Because TypeParser inside of block "try" also could throw Exceptions.

File: User_Story#1/SortingApp/SortinfApp.UnitTests/SortingAlgorithms/InsertionSorterTest.cs (hunk 4 above)
Comment (AlEkSunder) [reply User_Story#1/SortingApp/SortinfApp.UnitTests/SortingAlgorithms/InsertionSorterTest.cs:39]: Agree.

File: User_Story#1/SortingApp/SortinfApp.UnitTests/Utilities/Validator/LocalFileValidatorTest.cs (hunk 8 above)
Comment (AlEkSunder) [reply User_Story#1/SortingApp/SortinfApp.UnitTests/Utilities/Validator/LocalFileValidatorTest.cs:70]: Lets imagine a test-method that checks if a data exists.
In the method we specified a path that is not valid.
In that case method will work.
But we will get an answer about 'Does logic work correct if the path is incorrect'
And we will not get an answer about 'Does logic work correct when the data exists'.

So these are two different test-cases that need to be checked separately.

File: User_Story#1/SortingApp/SortinfApp.UnitTests/Extensions/TypeParserTest.cs [hunk 27]
```diff
@@ -0,0 +1,100 @@
...
+        /// Tests ChangeType method if pass empty value it will throw ArgumentNullException.
+        /// </summary>
+        /// <owner>Anton Petrenko</owner>
+        [TestMethod]
+        public void TypeParser_ChangeType_PassEmptyValue_ThrowArgumentNullException()
+        {
+            //
+            // Arrange.
+            //
+            string value = "";
```
Comment (AlEkSunder) [inline User_Story#1/SortingApp/SortinfApp.UnitTests/Extensions/TypeParserTest.cs]: Do you remember about string.Empty?
Please use that instead.

File: User_Story#1/SortingApp/SortinfApp.UnitTests/Utilities/DataReaderTest.cs [hunk 28]
```diff
@@ -0,0 +1,119 @@
...
+        /// <owner>Anton Petrenko</owner>
+        [TestMethod]
+        public void DataReader_ReadContent_PassEmptyPathToTheFile_ThrowArgumentNullException()
+        {
+            //
+            // Arrange.
+            //
+            var validator = new StubIValidator()
+            {
+                IsDataExistString = (string str) => true
```
Comment (AlEkSunder) [inline User_Story#1/SortingApp/SortinfApp.UnitTests/Utilities/DataReaderTest.cs]: Just FYI, we can skip type declaration in lambda expression.
It should work in this way too:
IsDataExistString = (str) => true

But it is up to you to decide how to use that, perhaps some way is more clear for you.

File: User_Story#1/SortingApp/ViewModel/CollectionSortingViewModel.cs [hunk 29]
```diff
@@ -101,8 +101,9 @@ public string Message
             get { return this.message; }
             set
             {
-                this.message = value;
-                this.OnPropertyChanged(nameof(Message));
+                if (message != value)
+                    this.message = value;
+                    this.OnPropertyChanged(nameof(Message));
```
Comment (AlEkSunder) [inline User_Story#1/SortingApp/ViewModel/CollectionSortingViewModel.cs:108]: Please add an empty line after if block and align code on the line 106.
This code will not work correct.
You will notify UI each time.
correct code:

if(this.message == value)
   return;

this.message = value;
this.OnPropertyChanged(nameof(this.Message));

**Review by AlEkSunder — APPROVED** [review]
Please pay attention to the last my comment in the VM class.

File: User_Story#1/SortingApp/ViewModel/CollectionSortingViewModel.cs (hunk 29 above)
Comment (alevel-petrenko) [reply User_Story#1/SortingApp/ViewModel/CollectionSortingViewModel.cs:108]: 🤦‍♂️
